tests/components/webostv/test_media_player.py

```diff
@@ -1,56 +1,81 @@
 """The tests for the LG webOS media player platform."""
-import unittest
-from unittest import mock
+import sys
 
-from homeassistant.components.webostv import media_player as webostv
+import pytest
 
+from homeassistant.components import media_player
+from homeassistant.components.media_player.const import (
+    ATTR_INPUT_SOURCE,
+    ATTR_MEDIA_VOLUME_MUTED,
+    SERVICE_SELECT_SOURCE,
+)
+from homeassistant.components.webostv import DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_HOST,
+    CONF_NAME,
+    SERVICE_VOLUME_MUTE,
+)
+from homeassistant.setup import async_setup_component
 
-class FakeLgWebOSDevice(webostv.LgWebOSDevice):
-    """A fake device without the client setup required for the real one."""
+if sys.version_info >= (3, 8, 0):
+    from unittest.mock import patch
+else:
+    from asynctest import patch
 
-    def __init__(self, *args, **kwargs):
-        """Initialise parameters needed for tests with fake values."""
-        self._source_list = {}
-        self._client = mock.MagicMock()
-        self._name = "fake_device"
-        self._current_source = None
 
+NAME = "fake"
+ENTITY_ID = f"{media_player.DOMAIN}.{NAME}"
 
-class TestLgWebOSDevice(unittest.TestCase):
-    """Test the LgWebOSDevice class."""
 
-    def setUp(self):
-        """Configure a fake device for each test."""
-        self.device = FakeLgWebOSDevice()
+@pytest.fixture(name="client")
+def client_fixture():
+    """Patch of client library for tests."""
+    with patch(
+        "homeassistant.components.webostv.WebOsClient", autospec=True
+    ) as mock_client_class:
+        yield mock_client_class.return_value
 
-    def test_select_source_with_empty_source_list(self):
-        """Ensure we don't call client methods when we don't have sources."""
-        self.device.select_source("nonexistent")
-        assert 0 == self.device._client.launch_app.call_count
-        assert 0 == self.device._client.set_input.call_count
 
-    def test_select_source_with_titled_entry(self):
-        """Test that a titled source is treated as an app."""
-        self.device._source_list = {
-            "existent": {"id": "existent_id", "title": "existent_title"}
-        }
+async def setup_webostv(hass):
+    """Initialize webostv and media_player for tests."""
+    assert await async_setup_component(
+        hass, media_player.DOMAIN, {media_player.DOMAIN: {}},
```

Comment:
The media_player component will be set up automatically as a dependency of the webostv media_player platform. We can remove this call.